types/react-use-dimensions/index.d.ts

```diff
@@ -0,0 +1,21 @@
+type Dimensions = ReturnType<typeof HTMLElement.prototype.getBoundingClientRect> & {
+    x: number;
+    y: number;
+    top: number;
+    left: number;
+};
+
+interface Args {
+    liveMeasure?: boolean;
+}
+
+declare function useDimensions<
+    // eslint-disable-next-line @definitelytyped/no-unnecessary-generics
+    T extends HTMLElement = HTMLElement,
+>(args?: Args): [
+    (node: T) => void,
```

Comment:
Specifically, the type parameter is only used in one place, which means that it's equivalent to a type annotation on the arrow: `(node: HTMLElement) => void`. If somebody wants to specify a subtype of HTMLElement, they don't do it like `useDimensions<MyElement>(node => ...`, they do it like so `useDimensions((node: MyElement) => ...`

```suggestion
declare function useDimensions(args?: Args): [
    (node: HTMLElement) => void,
```